### ISSUE TITLE
How to get the BLEU score from saved models for scaling NMT?

### ISSUE THREAD

**Issue by asispatra** [pytorch/fairseq#617]
How to get the BLEU score from saved models or checkpoints for scaling NMT paper? 
I have used WMT16 En-De dataset for training.

Thanks.

**Comment by huihuifan** [comment]
we provide many examples for generation commands and you can check the docs

https://github.com/pytorch/fairseq/tree/master/examples/translation

**Comment by asispatra** [comment]
@huihuifan ,

Any idea why I am getting BLEU score 0.00 ?
```
$ fairseq-generate <Test_Data> --path checkpoints/checkpoint_<[0-9]>.pt --beam 5 --batch-size 128 --remove-bpe | tee /tmp/gen.out
.
.
.

| Translated 3003 sentences (87069 tokens) in 25.4s (118.37 sentences/s, 3432.06 tokens/s)
| Generate test with beam=5: BLEU4 = 0.00, 1.0/0.0/0.0/0.0 (BP=1.000, ratio=1.000, syslen=3003, reflen=3003)
$ grep ^H /tmp/gen.out | cut -f3- > /tmp/gen.out.sys
$ grep ^T /tmp/gen.out | cut -f2- > /tmp/gen.out.ref
$ fairseq-score --sys /tmp/gen.out.sys --ref /tmp/gen.out.ref
Namespace(ignore_case=False, order=4, ref='/tmp/gen.out.ref', sacrebleu=False, sys='/tmp/gen.out.sys')
BLEU4 = 0.00, 1.0/0.0/0.0/0.0 (BP=1.000, ratio=1.000, syslen=3003, reflen=3003)
```

Thanks.

**Comment by ToWeRT1An** [comment]
have u fixed the problem?

> @huihuifan ,
> 
> Any idea why I am getting BLEU score 0.00 ?
> 
> ```
> $ fairseq-generate <Test_Data> --path checkpoints/checkpoint_<[0-9]>.pt --beam 5 --batch-size 128 --remove-bpe | tee /tmp/gen.out
> .
> .
> .
> 
> | Translated 3003 sentences (87069 tokens) in 25.4s (118.37 sentences/s, 3432.06 tokens/s)
> | Generate test with beam=5: BLEU4 = 0.00, 1.0/0.0/0.0/0.0 (BP=1.000, ratio=1.000, syslen=3003, reflen=3003)
> $ grep ^H /tmp/gen.out | cut -f3- > /tmp/gen.out.sys
> $ grep ^T /tmp/gen.out | cut -f2- > /tmp/gen.out.ref
> $ fairseq-score --sys /tmp/gen.out.sys --ref /tmp/gen.out.ref
> Namespace(ignore_case=False, order=4, ref='/tmp/gen.out.ref', sacrebleu=False, sys='/tmp/gen.out.sys')
> BLEU4 = 0.00, 1.0/0.0/0.0/0.0 (BP=1.000, ratio=1.000, syslen=3003, reflen=3003)
> ```
> 
> Thanks.

**Comment by myleott** [comment]
What model are you using? Please start with one of the pretrained models to make sure your generation command is correct: https://github.com/pytorch/fairseq/tree/master/examples/scaling_nmt

Later, when training your own models, you need to make sure the training/valid loss is good, otherwise your model will not produce sensible translations.
